multimodal/src/autogluon/multimodal/learners/default_learner.py

```diff
@@ -0,0 +1,2906 @@
+            name="",
+            version="",
+        )
+
+        ## Moved to the begining to join the setup steps for config
+        # # 22. training environment-dependent configs
+        # num_gpus = compute_num_gpus(config_num_gpus=config.env.num_gpus, strategy=config.env.strategy)
+
+        # precision = infer_precision(num_gpus=num_gpus, precision=config.env.precision)
+
+        # if num_gpus == 0:  # CPU only training
+        #     grad_steps = max(
+        #         config.env.batch_size // (config.env.per_gpu_batch_size * config.env.num_nodes),
+        #         1,
+        #     )
+        # else:
+        #     grad_steps = max(
+        #         config.env.batch_size // (config.env.per_gpu_batch_size * num_gpus * config.env.num_nodes),
+        #         1,
+        #     )
+
+        # if not hpo_mode:
+        #     if num_gpus <= 1:
+        #         if config.env.strategy == DEEPSPEED_OFFLOADING:  # Offloading currently only tested for single GPU
+        #             assert version.parse(pl.__version__) >= version.parse(
+        #                 DEEPSPEED_MIN_PL_VERSION
+        #             ), f"For DeepSpeed Offloading to work reliably you need at least pytorch-lightning version {DEEPSPEED_MIN_PL_VERSION}, however, found {pl.__version__}. Please update your pytorch-lightning version."
+        #             from .optimization.deepspeed import CustomDeepSpeedStrategy
+
+        #             strategy = CustomDeepSpeedStrategy(
+        #                 stage=3,
+        #                 offload_optimizer=True,
+        #                 offload_parameters=False,
+        #                 allgather_bucket_size=config.env.deepspeed_allgather_size,
+        #                 reduce_bucket_size=config.env.deepspeed_allreduce_size,
+        #             )
+        #         else:
+        #             strategy = None
+        #     else:
+        #         strategy = config.env.strategy
+        # else:
+        #     # we don't support running each trial in parallel without ray lightning
+        #     if use_ray_lightning:
+        #         strategy = hpo_kwargs.get("_ray_lightning_plugin")
+        #     else:
+        #         strategy = None
+        #         num_gpus = min(num_gpus, 1)
+
+        # config.env.num_gpus = num_gpus
+        # config.env.precision = precision
+        # config.env.strategy = strategy if not config.env.strategy == DEEPSPEED_OFFLOADING else DEEPSPEED_OFFLOADING
+        # self._config = config
+
+        ## TODO: Complete the save later. IGNORE THIS FOR NOW
+        # save artifacts for the current running, except for model checkpoint, which will be saved in trainer
+        # self.save(save_path, standalone=standalone)
+
+        blacklist_msgs = ["already configured with model summary"]
+        log_filter = LogFilter(blacklist_msgs)
+        # 23. Declare pl.Trainer
+        with apply_log_filter(log_filter):
+            trainer = self._get_trainer(
```

Comment:
To be more accurate, consider using `_init_lightning_trainer`. The callbacks and plugins are designed for lightning trainer.

Reply:
Also, make `_init_lightning_trainer ` cover the above log_filter logic.

Reply:
Done